Feature to master

balrom1981 opened the pull request.

balrom1981 committed:
model added

balrom1981 committed:
dto added

balrom1981 committed:
DBConnector added

balrom1981 committed:
mapper added

balrom1981 committed:
mapper added

balrom1981 committed:
repository added

balrom1981 committed:
service added

balrom1981 committed:
servlet added

balrom1981 committed:
descriptions added

balrom1981 committed:
repository tests added

balrom1981 committed:
service tests added

balrom1981 committed:
servlet tests added

balrom1981 committed:
files of creating db and tables added

balrom1981 committed:
add notification for user in method doDelete package servlet

danilskryl commented on src/main/java/ru/balrom/dto/PersonDto.java:
```diff
@@ -0,0 +1,36 @@
+@AllArgsConstructor
+@Builder
+@ToString
+
```
Лишний отступ

danilskryl commented on src/main/java/ru/balrom/mapper/ApartmentMapper.java:
```diff
@@ -0,0 +1,27 @@
+
+    @Override
+    public ApartmentDto toDto(Apartment apartment) {
+        if (apartment==null){
```
Форматирование

danilskryl commented on src/main/java/ru/balrom/mapper/ApartmentMapper.java:
```diff
@@ -0,0 +1,27 @@
+ * class ApartmentMapper предназначен для преобразования объекта типа Apartment в ApartmentDto
+ * и объекта типа ApartmentDto в Apartment
+ */
+
```
Лишний отступ

danilskryl commented on src/main/java/ru/balrom/mapper/ApartmentMapper.java:
```diff
@@ -0,0 +1,27 @@
+import ru.balrom.model.Apartment;
+
+/**
+ * class ApartmentMapper предназначен для преобразования объекта типа Apartment в ApartmentDto
```
Не нужно писать в доке class ApartmentMapper. Только то для чего он нужен

danilskryl commented on src/main/java/ru/balrom/service/ApartmentService.java:
```diff
@@ -0,0 +1,72 @@
+     */
+    @Override
+    public List<ApartmentDto> getAll() {
+        return apartmentRepository.getAll().stream().map(mapper::toDto).collect(Collectors.toList());
```
Лучше переносить на следующую строки команды стрима после .stream()

danilskryl commented on src/main/java/ru/balrom/servlet/CarServlet.java:
```diff
@@ -0,0 +1,169 @@
+    @Override
+    protected void doPut(HttpServletRequest request, HttpServletResponse response)
+            throws ServletException, IOException {
+        response.setContentType("application/json");
```
Лучше использовать константу

danilskryl commented on src/test/java/ru/balrom/repository/ApartmentRepositoryTest.java:
```diff
@@ -0,0 +1,118 @@
+        verify(preparedStatement).executeUpdate();
+
+    }
+
+
```
Лишние отступы

danilskryl commented on src/test/java/ru/balrom/repository/ApartmentRepositoryTest.java:
```diff
@@ -0,0 +1,118 @@
+    @BeforeEach
+    public void set() {
+        repository = new ApartmentRepository(connector);
+        apartmentExpected = Apartment.builder().id(ID).city(CITY).roomAmount(ROOM_AMOUNT).personId(PERSON_ID).build();
```
У билдера, как и стримов лучше тоже переносить методы на разные строки после builder()

danilskryl reviewed (approved):
Работа отличная, код чистый. Больше всего не нравится, конечно, форматирование и код стайл. Он у тебя не соблюдается даже между методами в рамках одного класса. Не забывай про CTRL ALT L. Работа зачтена

balrom1981 committed:
Initial commit on Lenovo

balrom1981 committed:
Method getAllByPersonId was added to Repository and Service
Method doGet in Class PersonServlet was changed, function of showing all estate (cars and apartments) of person was added
